lib/puttyhosts/puttyhosts.go

```diff
@@ -238,3 +238,40 @@ func FormatHostCAPublicKeysForRegistry(hostCAPublicKeys map[string][]string, hos
 	}
 	return registryOutput
 }
+
+// CheckAndSplitValidityKey processes PuTTY's "Validity" string key into individual list elements
+// and checks that its formatting follows the simple pattern "<hostname> || <hostname> || ..."
+// PuTTY uses a custom string format to represent what hostnames a given key should be trusted for.
+// See https://the.earth.li/~sgtatham/putty/0.79/htmldoc/Chapter4.html#config-ssh-cert-valid-expr for details.
+func CheckAndSplitValidityKey(input string, caName string) ([]string, error) {
+	var output []string
+	docsURL := "https://goteleport.com/docs/connect-your-client/putty/#troubleshooting"
+
+	// split the input string on spaces
+	splitTokens := strings.Split(input, " ")
+
+	// if the total number of hostnames and tokens doesn't equal an odd number, we can return an error early as the string is invalid
+	if len(splitTokens)%2 != 1 {
+		return nil, trace.BadParameter("validity string for %v contains non-odd [%d] number of entries, see %v", caName, len(splitTokens), docsURL)
+	}
+	for count, token := range splitTokens {
```

Comment:
nit: I'd prefer to name it `i` or `index` instead of `count`, just a bit more conventional and because it's a bit weird for a "count" to start at 0